### PR TITLE
Expose $TTL as an option

### LINKED ISSUE #7: Expose defaults.ttl as an option
It would be nice to be able to set the default TTL globally for a given zone.

### PR DESCRIPTION
**Pull request by ncfavier**
This PR exposes the `$TTL` directive ([RFC 2308, section 4](https://tools.ietf.org/html/rfc2308#section-4)) as a zone option `TTL` with default value `24 * 60 * 60`, and sets the default record `ttl` value to `null`, which causes it to be omitted and default to `$TTL`. This shouldn't change the semantics of existing zones.

I have tested this on my own DNS zone.

Closes #7

### CODE REVIEW

File: dns/types/record.nix [hunk 1]
```diff
@@ -40,12 +35,19 @@ let
     let
       data' =
         if isString data && rsubt ? fromString
-        then defaults // rsubt.fromString data
+        then (recordType rsubt).merge [] [ { file = ""; value = rsubt.fromString data; } ]
```
Comment (kirelagin) [inline dns/types/record.nix:38]: I’m not sure I understand this line. How is this different from simply `rsubt.fromString data`? 🤔

File: dns/types/zone.nix [hunk 2]
```diff
@@ -56,8 +56,14 @@ let
       concatStringsSep "\n\n" groups'
       + optionalString (sub != "") ("\n\n" + sub);
 
-  zone = types.submodule ({name, ...}: {
+  zone = types.submodule ({ name, ... }: {
     options = {
+      TTL = mkOption {
+        type = types.ints.unsigned;
+        default = 24 * 60 * 60;
+        example = 60 * 60;
```
Comment (kirelagin) [inline dns/types/zone.nix:64]: Hmm, maybe make this a `literalExample`?

**Review by kirelagin — COMMENTED** [review]
👍🏻 I like this idea!

File: dns/types/record.nix (hunk 1 above)
Comment (ncfavier) [reply dns/types/record.nix:38]: It adds the default values: `class`, `ttl`, but also whatever might be defined in the record type.

File: dns/types/zone.nix (hunk 2 above)
Comment (ncfavier) [reply dns/types/zone.nix]: Done.

**Comment by kirelagin** [comment]
@ncfavier Would you care to add yourself to the copyright notices (as a separate line) in the two files that you changed, please?

File: dns/types/record.nix (hunk 1 above)
Comment (kirelagin) [reply dns/types/record.nix:37]: Ahhhh, ok, I see now! Thanks for explaining. I guess this is really worth a comment in the code...

(I believe, the fact that a hack like this is needed is entirely my fault and i a consequence of my terrible abuse of the module system to get a nice DSL. I would really love to do it differently, but it is so fragile that I am scared. The plan is to add good tests, and then try to rewrite the internals to make hacks and boilerplate in the definition of records unnecessary.)

Comment (ncfavier) [reply dns/types/record.nix]: Added a comment.

Well, the module system could support this easily by letting us define a function to be applied to option values before merging them, like a preprocessing version of `apply`.

**Comment by ncfavier** [comment]
Done.

**Comment by kirelagin** [comment]
Thanks a lot! 🤝

**Pull request merged**